skimage/measure/_regionprops.py

```diff
@@ -358,22 +358,73 @@ def __eq__(self, other):
         return True
 
 
+def to_dict(regions, want=['label', 'bbox'], separator='-', cancel_auto=False):
+    """Convert image regions properties into a dictionary
+
+    Parameters
+    ----------
+    regions : (N,) list
+        _RegionProperties
+    want : list, optional
+        str of a property the user wants to include
+    separator : str, optional
+        used in joining coordinates
+    cancel_auto : bool, optional
+        prevents autmatic adding of label and bbox
+
+
+    Output
+    ------
+    A dictionary with properties in accordance to the wanted properties
+
+    Examples
+    --------
+    >>> from skimage import data, util
+    >>> from skimage.measure import label
+    >>> from pandas.DataFrame import from_dict
+    >>> img = util.img_as_ubyte(data.coins()) > 110
+    >>> label_img = label(img, connectivity=img.ndim)
+    >>> props = regionprops(label_img)
+    >>> out = from_dict(to_dict(props))
+
+
+"""
+
+    objects = {'image', 'coords', 'convex_image', 'filled_image',
+               'intensity_image'}
+
+    if not cancel_auto and 'label' not in want:
+        want.insert(0, 'label')
+    if not cancel_auto and 'bbox' not in want:
+        want.insert(0, 'bbox')
+
+    out = {}
+    arr = len(regions) * [0]
```

Comment:
Move this inside the `for prop in want`, and create a NumPy array rather than a list.

We should make a global dictionary mapping property name to column type. e.g.

```python
{'image': 'object', 'inertia_tensor': 'float', 'label': 'int', 'centroid': 'float', 'bbox': 'int', ...}
```

Then, inside the loop:

```python
for prop in properties:
    dtype = COL_DTYPES[prop]
    column = np.zeros(len(propslist), dtype=dtype)
```

Reply:
Possibly that is a bug! I suggest adding it and seeing if anything breaks in the tests. =)